frontend/src/home/graph.js

```diff
@@ -4,13 +4,20 @@ import './graph.css';
 
 import D3Graph from './graph-d3';
 
-class Graph extends React.PureComponent {
+class Graph extends React.Component {
     constructor(props) {
         super(props);
+        this.divRef = React.createRef();
         this.graph = React.createRef();
         this.slider = React.createRef();
-        this.divRef = React.createRef();
-        this.d3Graph = new D3Graph();
+        this.d3Graph = new D3Graph.D3Graph();
+    }
+
+    static defaultProps = {
+        callbacks: {loaded: () => {}, mouseenter: () => {}, mouseleave: () => {}, select: () => {}, deselect: () => {}},
```

Comment:
Could you leave a quick code comment explaining just what each callback receives. I saw deselect receives nothing, whereas select receives some data object. This is hard to figure out from a distance (or a year from now ;)